A) Implements a complete batch-download of selected files

noVaSon opened the pull request:
Highlights:
• add download Button to SoundInfoModal
• submit freesound-ids to the flask backend
• download the files and the metadata there and put it in a zip archive
• send it back to the frontend and save to disk
• remove the temporary files downloaded from the server with a delayed cleanup Thread

Adresses #38

noVaSon committed:
Copies V01 Selection behaviour from soundlist branch

noVaSon committed:
Merge branch 'master' of https://github.com/noVaSon/freesound-explorer into BatchDownload

noVaSon committed:
Merge branch 'master' into BatchDownload

noVaSon committed:
Simple per file download of the original - works well when user is logged in

noVaSon committed:
WIP starts to care for the backend side of batch download

noVaSon committed:
WIP more on frontend batch download

noVaSon committed:
Merge branch 'master' of https://github.com/ffont/freesound-explorer into BatchDownload

noVaSon committed:
Correctly passes arguments to batchDownloadSelectedOriginals

noVaSon committed:
Reaches backend for the first time

noVaSon committed:
Saves a wav loaded from the backend in the frontend

noVaSon committed:
WIP trying to eliminate paths in zip file and re-enabling send_file (wrong path)

noVaSon committed:
Enables batch download of multiple selected files as a zip file

noVaSon committed:
Considers click without selection and cleans up some logging

noVaSon committed:
WIP deletes downloaded files, not working yet

noVaSon committed:
Removes downloaded files and tempfolder from the server after timeout using a separate thread

noVaSon committed:
Improves some minor comments

noVaSon committed:
Enables system messages for batch download and creates the audio folder in the backend

noVaSon committed:
Adds a csv metadata file to the zip archive

noVaSon committed:
Cleans a comment

noVaSon committed:
Merge branch 'master' of https://github.com/ffont/freesound-explorer into BatchDownload

noVaSon committed:
Adds download from sound info modal

noVaSon committed:
Cleanup for PR

noVaSon committed:
Eliminates a lonesome console.log

noVaSon commented:
Some notes to myself:

• csv should be optimized (only selected columns, reshape lists (now: [u'tag1, u'tag2']))
• filenames should be identical with the entry in the csv
• parts of the python implementation should be modularized

noVaSon committed:
Reshapes csv file

noVaSon committed:
Changes audio filenames to origninal ones

ffont commented:
Wow, this looks like big news!
I'll be away for a week approx so I won't be able to review. I'll have a look after that.
In this way you also have time to add further improvements if needed.
Thanks!

noVaSon commented:
Thank you for this note @ffont!

Seems like we had similar temperatures this summer (37°C in Berlin this week!) :)
Enjoy your vacation!!

noVaSon committed:
Refactors into a FsZipper class and adds the audio temp folder to the settings.py

noVaSon committed:
Refactors into a FsZipper class and adds the audio temp folder to the settings.py

noVaSon commented:
Ok, ready for a merge :D

ffont commented on backend/fszipper.py:
```diff
@@ -0,0 +1,109 @@
+from backend.settings import DOWNLOAD_TEMP
```
`DOWNLOAD_TEMP` need to be added in `settings.example.py`.

ffont replied:
And maybe rename it to `DOWNLOAD_FOLDER_PATH` for consistency with other settings.

ffont commented on src/containers/Sidebar/actions.js:
```diff
@@ -9,3 +16,21 @@ export const toggleSidebarVisibility = makeActionCreator(TOGGLE_SIDEBAR_VISIBILI
+  selectedSounds.forEach(soundID => fsIds.push(`${sounds[soundID].id.split('-')[0]}`));
+
+  if (fsIds.length) {
+    dispatch(displaySystemMessage(`Prepare download of ${selectedSounds.length} Sounds, please wait...`, MESSAGE_STATUS.INFO));
```
"Preparing download of..." better than "Prepare download of..."

ffont reviewed (changes requested):
Great work! Here are some comments:

- If not logged in, download fails with 500 error (unauthorized). Maybe this could be prevented by not allowing to download if user not logged in.
- I'd use the same download icon as in the sound panel as it is easier to understand (IMO).
- Also the download icon should disappear when no sounds are selected. 
- I'd add a badge with the number of selected sounds on top of the icon so things become even clearer.
- Building the CSV sometimes fails when trying to download sounds with non ascii characters in their description or name. For example downloading this one fails: https://freesound.org/people/sijeh/sounds/394561/
- Also maybe there is no need to add that much data in the CSV. Could be something simpler like the file we add when a pack is downloaded in Freesound (mainly license/user info)

noVaSon committed:
Addresses requested changes

renamed Download folder
added it to settings.example.py
made download logo disappear
added number of selected sounds batch

unicode issues have still to be solved

noVaSon commented:
News:

I adressed your issues @ffont  - exept the unicode problem creating the .csv
I'll take care of this the next few days.

Best!
Eric

noVaSon committed:
Corrects the method of setting the download path like requested

noVaSon committed:
Refactors system message like requested

noVaSon committed:
Finalizes PR

- eliminates unicode issues in download csv
- renames the audio files for better representation in DAWs
- positions the batch download button
- refactors the csv columns negative list into settings.py

noVaSon reviewed (commented):
Helpful suggestions! Than you!
It is ready to merge and deploy now.

noVaSon committed:
Cleans up comments, some styling

ffont commented:
Hi,
I tried it again, it works almost perfect now, but still I have a couple of comments:

- When not logged in, the download still fails. If clicking on download with the user not being logged in we should show a message that this action requires login.
- When selecting multiple sounds (shit + click) I would prevent the info window from opening (for the sound being currently selected). This is not really part of this PR but I realized it while trying it for downloading sounds and to me it makes sense. Basically the idea is that when a sound is selected we show the info window unless shit key is pressed (does this make sense to you?).
- I'd only show the download option when more than 1 sound is selected. Otherwise there is already a download button in the info window of the selected sound. We prefer users to use that method because they'll download directly from Freesound instead of Freesound Explorer downloading from Freesound and then serving to users.
- You'll have to add instructions about the download feature in the README.

Thanks for the work, it's a really nice feature!

noVaSon committed:
Modifies behaviour of the batch download button

shows only if selected sounds > 1
suggests to login via systemMessage

noVaSon committed:
Disables the SoundInfoModal when pressing the Multiselection Key

noVaSon committed:
Updates the README.MD

no pictures due to lacking access to cloud.githubusercontend.com

noVaSon commented:
@ffont 

I did it!
Much better now, you are right!

Can you please tell me how to access githubusercontent for uploading pictures though?

ffont commented:
> Can you please tell me how to access githubusercontent for uploading pictures though?

Hah, I think I played a trick which is to upload an image by dragging it into an issue or PR comment (here!) and then copying the URL. Just trying it now again:

![364911_s_a33eb96f_freesoundbeastwhoosh](https://user-images.githubusercontent.com/478615/44535588-a54b2400-a6fa-11e8-844e-e44a64876e82.jpg)

You drag an image in this editor and Github uploads it and creates a URL than then I used in the readme. Then I guess I deleted the comments. Hacky!

noVaSon committed:
Updates README.MD

noVaSon commented:
@ffont > Thanks for the hacky trick 🍡 
I'm DonE!
